tools/erc-repository-indexer/erc-contract-indexer/src/services/config.ts

```diff
@@ -0,0 +1,127 @@
+/*-
+ *
+ * Hedera Smart Contracts
+ *
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+import constants from '../utils/constants';
+import { ContractScannerService } from './contractScanner';
+import { Helper } from '../utils/helper';
+
+export class ConfigService {
+  /**
+   * @private
+   * @readonly
+   * @property {string} network - The network identifier for the Hedera network (e.g., testnet, mainnet).
+   */
+  private readonly network: string;
+
+  /**
+   * @private
+   * @readonly
+   * @property {string} startingPoint - The starting point for indexing, which can be a contract ID or a pagination pointer.
+   */
+  private readonly startingPoint: string;
+
+  constructor() {
+    this.network = process.env.HEDERA_NETWORK || '';
+    this.startingPoint = process.env.STARTING_POINT || '';
+    this.validateConfigs();
+
+    console.log(`Indexing process initiated. Target network: ${this.network}`);
+  }
+
+  /**
+   * Validates the configuration values for network and starting point.
+   * Throws an error if the configurations are invalid.
+   * @throws {Error} If HEDERA_NETWORK or STARTING_POINT is not properly configured.
+   */
+  private validateConfigs(): void {
+    if (!this.network || !constants.NETWORK_REGEX.test(this.network)) {
+      throw new Error(
+        `HEDERA_NETWORK Is Not Properly Configured: network=${this.network}`
+      );
+    }
+
+    if (
+      this.startingPoint &&
+      !constants.STARTING_POINT_REGEX.test(this.startingPoint)
+    ) {
+      throw new Error(
+        `STARTING_POINT Is Not Properly Configured: startingPoint=${this.startingPoint}`
+      );
+    }
+  }
+
+  /**
+   * Resolves the starting point for indexing based on the configuration.
+   * @returns {Promise<string | null>} A promise that resolves to the starting point string or null if not set.
+   */
+  async resolveStartingPoint(): Promise<string | null> {
+    if (constants.NEXT_POINTER_REGEX.test(this.startingPoint)) {
```

Comment:
I think there's a missing test for this flow.
Can you confirm and add one if it's indeed missing

Reply:
Ha absolutely great catch! I indeed missed the test for this one flow. Hmm how strange!

Anywho just updated and add a new UT for the flow. Thanks for pointing it out!